File: server/src/main/java/de/davelee/trams/server/request/AdjustSimulationIntervalRequest.java
package de.davelee.trams.server.request;

/**
 * This class is part of the TraMS Server REST API. It represents a request to adjust the simulation interval
 * by either increasing or decreasing the simulation interval for a particular company.
 * @author Dave Lee
 */
public class AdjustSimulationIntervalRequest {

    /**
     * The name of the company to adjust the satisfaction rate.
     */
    private String company;

    /**
     * The new value of the simulation interval in minutes.
     */
    private int simulationInterval;

    public AdjustSimulationIntervalRequest() {
    }

    public AdjustSimulationIntervalRequest(String company, int simulationInterval) {
        this.company = company;
        this.simulationInterval = simulationInterval;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public int getSimulationInterval() {
        return simulationInterval;
    }

    public void setSimulationInterval(int simulationInterval) {
        this.simulationInterval = simulationInterval;
    }

    @Override
    public String toString() {
        return "AdjustSimulationIntervalRequest{" +
                "company='" + company + '\'' +
                ", simulationInterval=" + simulationInterval +
                '}';
    }
}

File: desktop/src/app/options/adjustSimulationInterval.request.ts
/**
 * This class defines a model for requests to adjust the simulation interval in TraMS.
 */
export class AdjustSimulationIntervalRequest {

    private company: string;
    private simulationInterval: number;

    /**
     * Construct a new request to adjust the simulation interval.
     * @param company the company that we should adjust the difficulty level for.
     * @param simulationInterval the new simulation interval in minutes that should be set.
     */
    constructor(company: string, simulationInterval: number) {
        this.company = company;
        this.simulationInterval = simulationInterval;
    }
}